PowerShellTools/DebugEngine/DebugServiceEventsHandlerProxy.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+using System.ServiceModel;
+using System.Text;
+using System.Threading.Tasks;
+using PowerShellTools.Common.ServiceManagement.DebuggingContract;
+
+namespace PowerShellTools.DebugEngine
+{
+    /// <summary>
+    /// Proxy of debugger service event handlers
+    /// This works as InstanceContext for debugger service channel
+    /// </summary>
+    [CallbackBehavior(ConcurrencyMode = ConcurrencyMode.Multiple, UseSynchronizationContext = false)]
+    public class DebugServiceEventsHandlerProxy : IDebugEngineCallback
+    {
+        private ScriptDebugger _debugger;
+
+        public DebugServiceEventsHandlerProxy(){}

```

Comment:
Since I have overloaded constructor defined(for unit testing purpose), you cannot omit the default constructor.

From: Huanhuan Sun [mailto:notifications@github.com]
Sent: Wednesday, January 28, 2015 1:20 PM
To: Microsoft/poshtools
Cc: Eric Deng
Subject: Re: [poshtools] Outproc(64bit) local debugging (#62)


In PowerShellTools/DebugEngine/DebugServiceEventsHandlerProxy.cs<https://github.com/Microsoft/poshtools/pull/62#discussion_r23724589>:

> +using System.Text;

> +using System.Threading.Tasks;

> +using PowerShellTools.Common.ServiceManagement.DebuggingContract;

> +

> +namespace PowerShellTools.DebugEngine

> +{

> +    /// <summary>

> +    /// Proxy of debugger service event handlers

> +    /// This works as InstanceContext for debugger service channel

> +    /// </summary>

> +    [CallbackBehavior(ConcurrencyMode = ConcurrencyMode.Multiple, UseSynchronizationContext = false)]

> +    public class DebugServiceEventsHandlerProxy : IDebugEngineCallback

> +    {

> +        private ScriptDebugger _debugger;

> +

> +        public DebugServiceEventsHandlerProxy(){}

Can be removed

—
Reply to this email directly or view it on GitHub<https://github.com/Microsoft/poshtools/pull/62/files#r23724589>.